### ISSUE TITLE
copy summarized data to our S3 bucket for archiving

### ISSUE THREAD

**Issue by infinitewarp** [project-koku/koku#1067]
## User Story
As a Koku user, I want to get Koku's summarized data from any recorded past time, not just the last month.

This follows a similar path to https://github.com/project-koku/koku/issues/1024. We need to save the summarized data to our S3 bucket for long-time storage since it is periodically truncated from the SQL database. Then when the user requests an export, we can reference those archived copies in our S3 bucket.

## Assumptions and Questions
- Files should be stored hierarchically with the account ID, provider type, year, month, and (optionally) day in the S3 bucket using a pattern like:
  - /data_archive/acct_12345/aws/2019/08/01/tablename.csv
- If the data is not specifically relevant to individual days, it may be stored at the month directory level.
- Use the same path, bucket, etc. variables as defined for https://github.com/project-koku/koku/issues/1024
- We expect the summarized data to be periodically overwritten as new data is processed periodically by the celery tasks.
- See also: https://redhatmbu.aha.io/epics/COST-E-97
- The code that performs this query and saves the file(s) does not need to run on the same pod/machine that handles the raw files since this operation simply needs to query the database. Therefore it could be a truly async task that follows `Orchestrator.prepare`'s call to `summarize_reports`; we can probably call it at the end of `summarize_reports`.

## Acceptance Criteria

- [ ] Files for different accounts, provider types, and dates are stored in different directories following the described pattern.
- [ ] A function exists and is routinely called to query the database and write the summarized data to S3.

**Comment by lcouzens** [comment]
Verified.
Commit: 96a12fc0e55ee08d444f5919b549ec575683a988

**Comment by lcouzens** [comment]
I have noticed some potential issues with this, firstly I am seeing missing acct data from our CI bucket. I since hit the upload endpoint here `api/cost-management/v1/upload_normalized_data/` to try and get the other data uploaded but it seemed to fail. Checking the logs I saw the process being killed.

[2019-10-23 12:43:18,840: ERROR/MainProcess] Process 'ForkPoolWorker-3' pid:341 exited with 'signal 9 (SIGKILL)'

After a brief discussion with @infinitewarp pushing this back to review for investigation.
